### ISSUE TITLE
Plot transition matrix and histograms of ranking scores/ranking probabilities

### ISSUE THREAD

**Issue by cdawei** [chengsoonong/digbeta#55]
Plot POI transition matrix as image, plot the histogram of POI ranking scores and the ranking probabilities after Platt scaling.

**Comment by cdawei** [comment]
Solved by commit 19ea7746c7a0d517c20c4f0b67cf310eede1f722

**Comment by lexingxie** [comment]
include platt scaling curve in the middle plot
also break down the scores for positive and negative training data 

![image](https://cloud.githubusercontent.com/assets/2434228/12965631/c395a034-d0ab-11e5-875c-9d355cd019df.png)

**Comment by lexingxie** [comment]
think about how to train with imbalanced neg/pos examples. 
e.g. the left plot above. 

talk to people, and post one or more ideas here asap.

**Comment by cdawei** [comment]
re: the unbalanced neg/pos examples

Talked to Suvash, he pointed out one generally weight the loss of neg/pos examples during training, which is equivalent to over sampling the minority.

$\text{Loss} = \text{Loss}_\+ \times w_\+ + \text{Loss}_\- \times w_-$

where $\text{Loss}_+$ (and $\text{Loss}_-$) is total loss of positive (and negative) examples,
weight parameters are $w_+ = \frac{\text{#neg_example}}{\text{total_num_example}}$ and $w_- = \frac{\text{#pos_example}}{\text{total_num_example}}$.

**Comment by cdawei** [comment]
Commit 96c181bd244172890e35d657a238fc735aeedf67 implements weighted loss for imbalance pos/neg examples.
